### DISCUSSION TITLE
Customizing DispatchingExecutorServiceFactory in EnvironmentBuilder

### DISCUSSION THREAD

**Discussion by laststem** [rabbitmq/rabbitmq-stream-java-client#543] [Q&A] [answered]
### Is your feature request related to a problem? Please describe.

I want to check metrics between Netty Inboud handler and Frame Handler execution.

For example,
1. Consumer started
2. Message was published to the stream
3. Broker sent message to the consumer
4. [StreamHandler.channelRead](https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/v0.15.0/src/main/java/com/rabbitmq/stream/impl/Client.java#L2743) was called
5. FrameHandler task was [submitted](https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/v0.15.0/src/main/java/com/rabbitmq/stream/impl/Client.java#L2771) to dispatchingExecutorService
6. When the time comes, dispatchingExecutorService will execute the task.

In this situation, there are latency between 5 ~ 6 that refers to the time waiting for actual execution but couldn't check.

### Describe the solution you'd like

1. Move ExecutorServiceFactory from stream.impl package to stream package.
2. Add ExecutorServiceFactory to the EnvironmentBuilder(StreamEnvironmentBuilder), then StreamEnvironmentBuilder.ClientParameters holds ExecutorServiceFactory reference.
3. create ExecutorServiceFactory that can provide metrics. (This must be implemented by the library user)
4. Add it to EnvironmentBuilder

Is this a possible scenario? Or is there another way to measure it?

Thank you.

**Comment by acogoluegnes** [comment, marked as answer]
I'd suggest doing some profiling before changing the public API.
